File: Leet1578.ts
function minCost(colors: string, neededTime: number[]): number {
    let mxCol = colors[0];
    let mxTime = neededTime[0];

    let time = 0;
    for (let i = 1; i < colors.length; i++) {
        if (colors[i] != mxCol) {
            mxCol = colors[i];
            mxTime = neededTime[i];
        } else {
            if (mxTime < neededTime[i]) {
                time += mxTime;
                mxTime = neededTime[i];
            } else {
                time += neededTime[i];
            }
        }
    }

    return time;
};

File: Leet1578.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        char maxCol= colors.charAt(0);
        int maxTime = neededTime[0];

        int time = 0;
        for (int i = 1; i < colors.length(); i++) {
            if (maxCol != colors.charAt(i)) {
                maxCol = colors.charAt(i);
                maxTime = neededTime[i];
            } else {
                if (maxTime < neededTime[i]) {
                    time += maxTime;
                    maxTime = neededTime[i];
                } else {
                    time += neededTime[i];
                }
            }
        }

        return time;
    }
}